lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs

```diff
@@ -666,6 +695,31 @@ impl Client {
         Ok(vec![])
     }
 
+    fn tdp_sd_delete(
+        &mut self,
+        rdp_req: DeviceCloseRequest,
+        file: FileCacheObject,
+    ) -> RdpResult<Vec<Vec<u8>>> {
+        (self.tdp_sd_delete_request)(SharedDirectoryDeleteRequest {
+            completion_id: rdp_req.device_io_request.completion_id,
+            directory_id: rdp_req.device_io_request.device_id,
+            path: file.path,
+        })?;
+        self.pending_sd_delete_resp_handlers.insert(
+            rdp_req.device_io_request.completion_id,
+            Box::new(
+                |cli: &mut Self, res: SharedDirectoryDeleteResponse| -> RdpResult<Vec<Vec<u8>>> {
+                    if res.err_code == 0 {
+                        cli.prep_device_close_response(rdp_req, NTSTATUS::STATUS_SUCCESS)
+                    } else {
+                        cli.prep_device_close_response(rdp_req, NTSTATUS::STATUS_UNSUCCESSFUL)
+                    }
```

Comment:
```suggestion
                    let resp = if res.err_code == 0 { NTSTATUS::STATUS_SUCCESS } else { NTSTATUS::STATUS_UNSUCCESSFUL }
                    cli.prep_device_close_response(rdp_req, resp)
```

I prefer setting the code ahead of time and only have one call to `prep_device_close_response` - it's just a little easier to see that the control flow is the same and we're only toggling one parameter.